db_test.go

```diff
@@ -774,6 +776,189 @@ func TestCommitHooks_PreFails(t *testing.T) {
+		t.Fatal(err)
+	}
+	if c[0] != 1 {
+		t.Fatalf("we should have renamed the user to 'baz', and still find it")
+	}
+
+	// Query
+	row, err := db.Query("SELECT name FROM users WHERE name = ?", someTypeAlias("baz"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !row.Next() {
+		t.Fatal("there should be one row to read")
+	}
+	var name string
+	if err := row.Scan(&name); err != nil {
+		t.Fatal(err)
+	}
+	if name != "baz" {
+		t.Fatalf("we should have renamed the user to 'baz', and still be able to query it")
+	}
+	if err := row.Close(); err != nil {
+		t.Fatal(err)
+	}
+
+	// QueryRow
+	rowPtr := db.QueryRow("SELECT name FROM users WHERE name = ?", someTypeAlias("baz"))
+	if rowPtr == nil {
+		t.Fatal("there should be one row to read")
+	}
+	if err := rowPtr.Scan(&name); err != nil {
+		t.Fatal(err)
+	}
+	if name != "baz" {
+		t.Fatalf("we should have renamed the user to 'baz', and still be able to query it")
+	}
+}
+
+type someAge int
+
+type TypedUser struct {
+	ID   uint64        `db:"id"`
+	Name someTypeAlias `db:"name"`
+	Age  someAge       `db:"age"`
+}
+
+func TestTypeAlias_inMappedStruct(t *testing.T) {
+	db := makeTestDB(t, usersDDL)
+	defer db.Close()
+
+	if _, err := db.BindModel("users", TypedUser{}); err != nil {
+		t.Fatal(err)
+	}
+	e := &TypedUser{
+		ID:   0,
+		Name: someTypeAlias("foo"),
+		Age:  someAge(123),
+	}
+	if err := db.Insert(e); err != nil {
+		t.Fatal(err)
+	}
+
+	count := countUsersByName(t, db)
+
+	if count("foo") != 1 {
+		t.Fatalf("should start with one user, named 'foo'")
+	}
+
+	// Select
+	var users []TypedUser
+	if err := db.Select(&users, "SELECT * FROM users"); err != nil {
+		t.Fatal(err)
+	} else if len(users) != 1 {
+		t.Fatalf("should have one user, named 'foo'")
+	}
+	if string(users[0].Name) != "foo" {
+		t.Fatalf("our user should be named 'foo', was '%s'", users[0].Name)
+	}
+	if int(users[0].Age) != 123 {
+		t.Fatalf("our user should be aged 123 years, was %d", users[0].Age)
+	}
+
+	// Query
+	rows, err := db.Query("SELECT * FROM users")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !rows.Next() {
+		t.Fatalf("should have one user, named 'foo'")
+	}
+	var queriedUser TypedUser
+	if err := rows.StructScan(&queriedUser); err != nil {
+		t.Fatal(err)
+	}
+	if string(queriedUser.Name) != "foo" {
+		t.Fatalf("our user should be named 'foo', was '%s'", queriedUser.Name)
+	}
+	if int(queriedUser.Age) != 123 {
+		t.Fatalf("our user should be aged 123 years, was %d", queriedUser.Age)
+	}
+	rows.Close()
+
+	// QueryRow
+	row := db.QueryRow("SELECT * FROM users")
+	var queryRowUser TypedUser
+	if err := row.StructScan(&queryRowUser); err != nil {
+		t.Fatal(err)
+	}
+	if string(queryRowUser.Name) != "foo" {
+		t.Fatalf("our user should be named 'foo' years, was '%s'", queryRowUser.Name)
+	}
+	if int(queryRowUser.Age) != 123 {
+		t.Fatalf("our user should be aged 123 years, was %d", queryRowUser.Age)
+	}
+
+	// Get
```

Comment:
@jbowens btw, while testing on internal Square code, I noticed `Get` needed special handling and added it to this PR.